src/components/surveys/survey-design/question-edit/rhs-subcontrols/Year.tsx

```diff
@@ -1,78 +1,36 @@
-import {StyledFormControl} from '@components/surveys/widgets/SharedStyled'
 import AlertWithTextWrapper from '@components/widgets/AlertWithTextWrapper'
-import {SimpleTextInput, SimpleTextLabel} from '@components/widgets/StyledComponents'
-import {Box, Checkbox, FormControlLabel, Typography} from '@mui/material'
-import {theme} from '@style/theme'
+import {Box} from '@mui/material'
 import {FormatOptionsYear, YearQuestion} from '@typedefs/surveys'
-import React, {ChangeEvent} from 'react'
+import _ from 'lodash'
+import React from 'react'
+import YearRadioGroup, {YearFormatType} from './YearRadioGroup'
 
-const GeneralAllowCheckbox: React.FunctionComponent<{
-  value: boolean
-  type: 'PAST' | 'FUTURE'
-  onChange: (checked: boolean) => void
-}> = ({type, value, onChange}) => {
-  const CONFIG = {
-    PAST: {
-      label: 'Allow past years',
-      labelId: 'pastValueLbl',
-    },
-    FUTURE: {
-      label: 'Allow future years',
-      labelId: 'futureValueLbl',
-    },
-  }
+export const DEFAULT_MIN_YEAR = 1900
+export const DEFAULT_MAX_YEAR = new Date().getFullYear()
 
-  return (
-    <FormControlLabel
-      htmlFor={CONFIG[type].labelId}
-      sx={{mt: theme.spacing(1.5)}}
-      control={<Checkbox checked={value !== false} onChange={e => onChange(e.target.checked)} />}
-      label={<Typography sx={{fontWeight: '14px'}}>{CONFIG[type].label}</Typography>}
-    />
-  )
+const yearFormatToAllowValueMap: Record<YearFormatType, boolean | undefined> = {
+  SET: undefined,
+  ANY: true,
+  CURRENT: false,
```

Comment:
Create a mapping of the selected radio group value to the "allow" value, e.g. `allowPast`